src/Element/PluginSelect.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Drupal\commerce\Element;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\Core\Render\Element\FormElement;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @FormElement("commerce_plugin_select")
+ */
+class PluginSelect extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    return [
+      '#input' => TRUE,
+      '#plugin_type' => NULL,
+      '#title' => $this->t('Select plugin'),
+      '#process' => [
+        [$class, 'processPluginSelect'],
+        [$class, 'processAjaxForm'],
+      ],
+    ];
+  }
+
+  /**
+   * Process callback.
+   */
+  public static function processPluginSelect(&$element, FormStateInterface $form_state, &$complete_form) {
+    if (!$element['#plugin_type']) {
+      throw new \InvalidArgumentException('You must specify the plugin type ID.');
+    }
+
+    $element['#tree'] = TRUE;
+
+    /** @var \Drupal\Core\Executable\ExecutableManagerInterface $plugin_manager */
+    $plugin_manager = \Drupal::service('plugin.manager.' . $element['#plugin_type']);
+
+    $values = $element['#value'];
+
+    $target_id = !empty($values['target_id']) ? $values['target_id'] : '_none';
+
+    $ajax_wrapper_id = Html::getUniqueId('ajax-wrapper');
+    $ajax_settings = [
+      'callback' => [get_called_class(), 'pluginFormAjax'],
+      'wrapper' => $ajax_wrapper_id,
+      'options' => [
+        'query' => [
+          'element_parents' => implode('/', $element['#array_parents']),
+        ],
+      ],
+    ];
+
+    // Prefix and suffix used for Ajax replacement.
+    $element['#prefix'] = '<div id="' . $ajax_wrapper_id . '">';
+    $element['#suffix'] = '</div>';
+
+    $element['target_id'] = [
```

Comment:
This should have been target_plugin_id (and matcing target_plugin_configuration), for more clarity.

/me ducks thrown knives.